src/status_im/utils/ethereum/abi_spec.cljs

```diff
@@ -38,7 +38,7 @@
   (.hexToUtf8 utils (str "0x" x)))
 
 (defn hex-to-number [x]
-  (.hexToNumber utils (str "0x" x)))
+  (.toNumber (dependencies/Web3.prototype.toBigNumber (str "0x" x) 16)))
```

Comment:
the idea of this lib was to avoid using web3 this defeats the purpose

Reply:
Unfortunately `hexToNumber` doesn't work as we expected it. Also here it's only about using BigNumber (used elsewhere like money), not web3.

Reply:
I could create a util function to hide it away.

Reply:
just wondering why `hexToNumber` works different from our expectations, `utils` should provide functions to work with BN , so `(.toNumber (.toBN utils (str "0x" x) 16))`  should work

Reply:
@flexsurfer Just tried and it does not. It might be a BigNumber version issue.